src/Query/Builder.php

```diff
@@ -1205,10 +1238,18 @@ protected function compileWhereMonth(array $where): array
     {
         extract($where);
 
-        $where['operator'] = $operator;
-        $where['value'] = $value;
+        $value = (int) ltrim($value, '0');
 
-        return $this->compileWhereBasic($where);
+        return [
+            '$expr' => [
+                '$'.$operator => [
+                    [
+                        '$month' => '$'.$column,
```

Comment:
I assume `$column` is created by `extract($where)`. Is there any enforcement on that? It seems hard to track given we have no typing/shape information for the `$where` array.

Is it possible for `$column` to be a field path (e.g. `foo.bar`), or would it only be a top-level field name?

Reply:
Removing all this `extract` and replacing with array access would be more verbose but a lot more explicit.